### PR TITLE
MSI-2592: Alternative for IsProductSalableInterface when working with a list of items

### PR DESCRIPTION
**Pull request by nmalevanec**
### Description (*)

Deprecated IsProductSalableInterface, introduced new interfaces AreProudctsSalableInterface, IsProductSalableResultInterface and AreProductsSalableResultInterface. Replace IsproductSalableInterface usage with new AreProductsSalableInterface.
Notice: AreProudctsSalableInterface implementation still uses IsProudctSalableInterface implementation.

### Fixed Issues (if relevant)

1. https://github.com/magento/inventory/issues/2592 Alternative for IsProductSalableInterface when working with a list of items

### Manual testing scenarios (*)

1. Create orders with all available product types and verify no errors occurred.
2. Export and import all available product types and verify no errors occurred.
3. Verify API calls, including GraphQl and verify no errors occurred during requests.

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: InventorySalesApi/Api/Data/AreProductsSalableResultInterface.php [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+namespace Magento\InventorySalesApi\Api\Data;
+
+use Magento\Framework\Api\ExtensibleDataInterface;
+
+/**
+ * Represents result of "are products salable" interface.
+ *
+ * @api
+ */
+interface AreProductsSalableResultInterface extends ExtensibleDataInterface
```
Comment (sidolov) [inline InventorySalesApi/Api/Data/AreProductsSalableResultInterface.php:17]: I believe we can reuse `\Magento\InventorySalesApi\Api\Data\ProductsSalableResultInterface` here and remove the new interface

File: InventorySalesApi/Api/Data/IsProductSalableResultInterface.php [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+namespace Magento\InventorySalesApi\Api\Data;
+
+use Magento\Framework\Api\ExtensibleDataInterface;
+
+/**
+ * Represents "is product salable" result interface.
+ *
+ * @api
+ */
+interface IsProductSalableResultInterface extends ExtensibleDataInterface
```
Comment (sidolov) [inline InventorySalesApi/Api/Data/IsProductSalableResultInterface.php:17]: Please, reuse `\Magento\InventorySalesApi\Api\Data\ProductSalableResultInterface`

File: InventorySalesApi/Api/AreProductsSalableInterface.php [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+interface AreProductsSalableInterface
+{
+    /**
+     * Get are products salable for given SKUs and given Stock.
+     *
+     * @param string $skus in format "sku1,sku2,..."
+     * @param int $stockId
+     * @return \Magento\InventorySalesApi\Api\Data\AreProductsSalableResultInterface
+     */
+    public function execute(string $skus, int $stockId): AreProductsSalableResultInterface;
```
Comment (sidolov) [inline InventorySalesApi/Api/AreProductsSalableInterface.php:26]: Reuse `\Magento\InventorySalesApi\Api\Data\ProductsSalableResultInterface` here

File: InventorySales/Model/IsProductSalable/IsProductsSalableResult.php [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+namespace Magento\InventorySales\Model\IsProductSalable;
+
+use Magento\InventorySalesApi\Api\Data\IsProductSalableResultExtensionInterface;
+use Magento\InventorySalesApi\Api\Data\IsProductSalableResultInterface;
+
+/**
+ * @inheritDoc
+ */
+class IsProductsSalableResult implements IsProductSalableResultInterface
```
Comment (sidolov) [inline InventorySales/Model/IsProductSalable/IsProductsSalableResult.php:16]: Should be removed

File: InventorySales/Model/AreProductsSalable/AreProductsSalableResult.php [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+namespace Magento\InventorySales\Model\AreProductsSalable;
+
+use Magento\InventorySalesApi\Api\Data\AreProductsSalableResultExtensionInterface;
+use Magento\InventorySalesApi\Api\Data\AreProductsSalableResultInterface;
+use Magento\InventorySalesApi\Api\Data\IsProductSalableResultInterface;
+
+/**
+ * @inheritDoc
+ */
+class AreProductsSalableResult implements AreProductsSalableResultInterface
```
Comment (sidolov) [inline InventorySales/Model/AreProductsSalable/AreProductsSalableResult.php:17]: Should be removed

File: InventorySales/Model/AreProductsSalable.php [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+    ) {
+        $this->isProductSalable = $isProductSalable;
+        $this->isProductSalableResultFactory = $isProductSalableResultFactory;
+        $this->areProductsSalableResultFactory = $areProductsSalableResultFactory;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function execute(string $skus, int $stockId): AreProductsSalableResultInterface
```
Comment (sidolov) [inline InventorySales/Model/AreProductsSalable.php:54]: Why @skus is not an array?

**Review by sidolov — CHANGES_REQUESTED** [review]

File: InventorySalesApi/Api/Data/IsProductSalableResultInterface.php (hunk 2 above)
Comment (nmalevanec) [reply InventorySalesApi/Api/Data/IsProductSalableResultInterface.php:17]: There is no getSku() method in \Magento\InventorySalesApi\Api\Data\ProductSalableResultInterface, therefore the only way to  understand to which product result applied is to set it in extension attributes. Is it a better option than to introduce new interface?

File: InventorySales/Model/AreProductsSalable.php (hunk 6 above)
Comment (nmalevanec) [reply InventorySales/Model/AreProductsSalable.php:54]: As far as I know, there is no ability to pass array via REST API for GET request.

File: InventorySalesApi/Api/Data/AreProductsSalableResultInterface.php (hunk 1 above)
Comment (nmalevanec) [reply InventorySalesApi/Api/Data/AreProductsSalableResultInterface.php:17]: The reason I've introduced new interface described bellow.

**Comment by nmalevanec** [comment]
see https://github.com/magento/inventory/pull/2910

**Pull request closed without merging**